contracts/AelinPool.sol

```diff
@@ -97,18 +105,31 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
         aelinDealLogicAddress = _aelinDealLogicAddress;
         aelinRewardsAddress = _aelinRewardsAddress;
 
-        if (_poolData.allowListAddresses.length > 0 || _poolData.allowListAmounts.length > 0) {
+        address[] memory tmpAllowListAddresses = _poolData.allowListAddresses;
+        uint256[] memory tmpAllowListAmounts = _poolData.allowListAmounts;
+
+        if (tmpAllowListAddresses.length > 0 || tmpAllowListAmounts.length > 0) {
             require(
-                _poolData.allowListAddresses.length == _poolData.allowListAmounts.length,
+                tmpAllowListAddresses.length == tmpAllowListAmounts.length,
                 "allowListAddresses and allowListAmounts arrays should have the same length"
             );
-            for (uint256 i = 0; i < _poolData.allowListAddresses.length; i++) {
-                allowList[_poolData.allowListAddresses[i]] = _poolData.allowListAmounts[i];
-                emit AllowlistAddress(_poolData.allowListAddresses[i], _poolData.allowListAmounts[i]);
+            for (uint256 i = 0; i < tmpAllowListAddresses.length; i++) {
+                allowList[tmpAllowListAddresses[i]] = tmpAllowListAmounts[i];
+                emit AllowlistAddress(tmpAllowListAddresses[i], tmpAllowListAmounts[i]);
             }
             hasAllowList = true;
         }
 
+        NftData[] memory tmpNftData = _poolData.nftData;
+
+        if (tmpNftData.length > 0) {
+            require(tmpNftData.length <= 5, "max 5 collections allowed");
+            for (uint256 i = 0; i < tmpNftData.length; i++) {
+                nftCollectionDetails[tmpNftData[i].collectionAddress] = tmpNftData[i];
```

Comment:
not sure where my other comment went but we definitely want to double check and make sure they are all ERC721/ cryptopunks ot ERC1155 but we dont want both ERC721 and ERC1155 being added to the same pool by the sponsor

Reply:
see 96e358261f8e04f73bb908935ba0db1146f14740